code/controllers/subsystem/job.dm

```diff
@@ -432,23 +433,18 @@ SUBSYSTEM_DEF(job)
 		if(length(GLOB.jobspawn_overrides[rank]))
 			S = pick(GLOB.jobspawn_overrides[rank])
 		if(S)
-			S.JoinPlayerHere(H, FALSE)
+			S.JoinPlayerHere(living_mob, FALSE)
 		if(!S) //if there isn't a spawnpoint send them to latejoin, if there's no latejoin go yell at your mapper
 			log_world("Couldn't find a round start spawn point for [rank]")
-			SendToLateJoin(H)
+			SendToLateJoin(living_mob)
 
 
-	if(H.mind)
-		H.mind.assigned_role = rank
+	if(living_mob.mind)
+		living_mob.mind.assigned_role = rank
 
+	to_chat(M, "<b>You are the [rank].</b>")
 	if(job)
-		var/new_mob = job.equip(H, null, null, joined_late , null, M.client)
-		if(ismob(new_mob))
-			H = new_mob
-			if(!joined_late)
-				N.new_character = H
-			else
-				M = H
```

Comment:
Silicons override equip() and return a new mob, which then replaces the old mob that was "robotized" and qdel'd.
[Shitcode](https://github.com/tgstation/tgstation/blob/master/code/modules/jobs/job_types/cyborg.dm#L17)
[Robotized](https://github.com/tgstation/tgstation/blob/master/code/modules/mob/transform_procs.dm#L389)
It really should be docced or refactored.

Reply:
kill me

Reply:
not on your life

Reply:
when this shitcode is in ashes, then you have my permission to die